gapic/ads-templates/%namespace/%name/%version/%sub/services/%service/transports/rest.py.j2

```diff
@@ -233,7 +292,7 @@ class {{service.name}}RestTransport({{service.name}}Transport):
             },
             {% endfor %}{# rule in method.http_options #}
             ]
-
+            request, metadata = self._interceptor.pre_{{ method.name|snake_case }}(request, metadata)
```

Comment:
At first I imagined we'd be able to pass a _list_ of interceptors that could be chained together, but understand that requires additional architecture here. We can handle the chaining within the body of the `pre_`/`post_` functions.

Reply:
Exactly, that's easily done by something like this:
```python
class InterceptorChainer:
    def __init__(self, pre_chain, post_chain):
        self.pre_chain = pre_chain
        self.post_chain = post_chain
        
    def __getattr__(self, name):
        if name.startswith("pre_"):
            def pre(self, request, metadata):
                for i in self.pre_chain:
                    request, metadata = getattr(i, name)(request, metadata)
                return request, metadata
                
            return pre
                
        elif name.startswith("post_"):
            def post(self, response):
                for i in self.post_chain:
                    response = getattr(i, name)(response)
                return response
            
            return post
            
        else:
            raise AttributeError(f"No such attribute: {name}")
```